### PR TITLE
New post: Intro to BDD with Gherkin

### PR DESCRIPTION
**Pull request by hilton**
New blog post draft for review

### CODE REVIEW

File: _posts/2017-05-03-bdd-gherkin.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+The second version described the same scenario, but focused on the specific business rule about which denominations to use:
+
+```gherkin
+Scenario: Dispense $50 cash in multiple denominations
+  When $50 is dispensed by the ATM
+  Then the following bills should be received:
+    | count | denomination |
+    | 2     | $20          |
+    | 1     | $10          |
+```
```
Comment (TimKam) [inline _posts/2017-05-03-bdd-gherkin.md]: Why should the system send out exactly these bills? _Dispense $50 cash in multiple denominations_ is less specific than the expected solution. I find that confusing.

File: _posts/2017-05-03-bdd-gherkin.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+For the first exercise, we discussed the pros and cons for each version of the scenario.
+Seb then cleverly tricked us into listing the benefits and disadvantages of each version, after which he explained why all of our observations about the long version were really disadvantages.
```
Comment (TimKam) [inline _posts/2017-05-03-bdd-gherkin.md]: Some examples would help here. Which disadvantages were perceived as advantages at first?

File: _posts/2017-05-03-bdd-gherkin.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+## Gherkin tips
+
+Another workshop exercise had us write scenarios in Gherkin for the public library example.
+Writing Gherkin feature descriptions turns out to be harder than it might look, given the simplicity of the result.
+At this point, Seb’s _Gherkin tips_ made a lot more sense:
+
+![Gherkin tips](../2017/gherkin-tips.jpg)
+
+As always, the thorny issue of naming appears, hence the first tip to use _Friends episode_ naming: _The one where the ATM dispenses cash_ or something.
```
Comment (TimKam) [inline _posts/2017-05-03-bdd-gherkin.md]: It took me some time to associate _Friends episode naming_ with the TV series _Friends_. But perhaps this is just me being TV illiterate.

File: _posts/2017-05-03-bdd-gherkin.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+## Next steps
+
+Towards the end of the workshop, I was getting more curious about [Cucumber](https://github.com/cucumber/cucumber), which it had become clear that we weren’t going to cover.
+As far as I can tell, Cucumber is a tool for executing Gherkin scenarios in the form of automated tests.
+It turns out that Seb co-wrote a book about Cucumber - [The Cucumber for Java Book](https://pragprog.com/book/srjcuc/the-cucumber-for-java-book), so I guess I know what I have to do next.
+
+<a title="The Cucumber for Java Book" href="https://pragprog.com/book/srjcuc/the-cucumber-for-java-book"><img style="border:1px solid #ccc" src="../2017/cucumber-for-java-book.jpg"></a>
+
+One more thing: at one point during the workshop, we encountered the suggestion that you can also use Gherkin to describe and test an API, rather than user-facing application functionality.
```
Comment (TimKam) [inline _posts/2017-05-03-bdd-gherkin.md]: _we encountered the suggestion_ is probably a bit wordy

File: _posts/2017-05-03-bdd-gherkin.md (hunk 1 above)
Comment (hilton) [inline _posts/2017-05-03-bdd-gherkin.md]: Unfortunately, Gherkin doens't seem to have a _Because_ keword :)

Is it at least clear that I’m quoting the example used in the workshop, rather than including my own example, so I can't really comment on why it’s exactly like this.

**Comment by hilton** [comment]
@TimKam I’ve resolved the review comments, except for the one I replied to.

File: _posts/2017-05-03-bdd-gherkin.md (hunk 1 above)
Comment (hilton) [inline _posts/2017-05-03-bdd-gherkin.md]: Perhaps this just means that the scenario name should be better. This is the kind of question that probably requires more experience with Gherkin.
